conda-store-server/conda_store_server/_internal/worker/tasks.py

```diff
@@ -153,33 +153,33 @@ def task_update_conda_channels(self):
 @shared_task(base=WorkerTask, name="task_update_conda_channel", bind=True)
 def task_update_conda_channel(self, channel_name):
     conda_store = self.worker.conda_store
-    with conda_store.session_factory() as db:
-        settings = conda_store.get_settings()
-
-        # sanitize the channel name as it's an URL, and it's used for the lock.
-        sanitizing = {
-            "https": "",
-            "http": "",
-            ":": "",
-            "/": "_",
-            "?": "",
-            "&": "_",
-            "=": "_",
-        }
-        channel_name_sanitized = channel_name
-        for k, v in sanitizing.items():
-            channel_name_sanitized = channel_name_sanitized.replace(k, v)
-
-        task_key = f"lock_{self.name}_{channel_name_sanitized}"
-
-        is_locked = False
-
-        if conda_store.config.redis_url is not None:
-            lock = conda_store.redis.lock(task_key, timeout=60 * 15)  # timeout 15min
-        else:
-            lockfile_path = os.path.join(f"/tmp/task_lock_{task_key}")
-            lock = FileLock(lockfile_path, timeout=60 * 15)
+    settings = conda_store.get_settings()
+
+    # sanitize the channel name as it's an URL, and it's used for the lock.
+    sanitizing = {
+        "https": "",
+        "http": "",
+        ":": "",
+        "/": "_",
+        "?": "",
+        "&": "_",
+        "=": "_",
+    }
+    channel_name_sanitized = channel_name
+    for k, v in sanitizing.items():
+        channel_name_sanitized = channel_name_sanitized.replace(k, v)
+
+    task_key = f"lock_{self.name}_{channel_name_sanitized}"
+
+    is_locked = False
+
+    if conda_store.config.redis_url is not None:
+        lock = conda_store.redis.lock(task_key, timeout=60 * 15)  # timeout 15min
+    else:
+        lockfile_path = os.path.join(f"/tmp/task_lock_{task_key}")
+        lock = FileLock(lockfile_path, timeout=60 * 15)
 
+    with conda_store.session_factory() as db:
```

Comment:
I guess if this line is being moved to the place where it is actually needed, it can go further down inside `if is_locked`?